### PR TITLE
envtest: retry on conflict while updating VRG

### PR DESCRIPTION
**Pull request by ELENAGER**
Add retry when update VRG fails because of modification conflict

Fixes: https://github.com/RamenDR/ramen/issues/1407

### CODE REVIEW

**Review by rakeshgm — APPROVED** [review]

File: controllers/vrg_volrep_test.go [hunk 1]
```diff
@@ -1515,6 +1516,18 @@ func (v *vrgTest) getVRG() *ramendrv1alpha1.VolumeReplicationGroup {
 	return vrgGet(v.vrgNamespacedName())
 }
 
+func updateVRG(vrg *ramendrv1alpha1.VolumeReplicationGroup) {
+	retryErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		vrgTmp := vrgGet(client.ObjectKeyFromObject(vrg))
+
+		vrgTmp.Spec = vrg.Spec
+
+		return k8sClient.Update(context.TODO(), vrgTmp)
```
Comment (nirs) [inline controllers/vrg_volrep_test.go:1525]: vrgTmp works, but maybe we can use more meaningful names?

```
func updateVRG(desired *ramendrv1alpha1.VolumeReplicationGroup) {
	retryErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
		current := vrgGet(client.ObjectKeyFromObject(desired))

		current.Spec = desired.Spec

		return k8sClient.Update(context.TODO(), current)
```

**Review by nirs — COMMENTED** [review]
Looks good, suggested nicer names.

**Review by nirs — APPROVED** [review]
Good enough as is, lets get rid of these failures.

File: controllers/vrg_volrep_test.go (hunk 1 above)
Comment (ELENAGER) [reply controllers/vrg_volrep_test.go]: @nirs Done

**Review by nirs — APPROVED** [review]

**Review by BenamarMk — APPROVED** [review]

**Pull request merged**